### PR TITLE
Automate Docker images build & push

### PR DESCRIPTION
**Pull request by joanlopez**
Contributes to https://github.com/grafana/xk6-kerberos/issues/35

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        run: |
+          docker build -t ghcr.io/grafana/gokrb5-test:${{ matrix.pair.tag }} testenv/docker/${{ matrix.pair.dir }}
+      - name: Log in to GitHub Container Registry
+        if: ${{ github.ref == 'refs/heads/master' }}
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login https://ghcr.io -u ${{ github.actor }} --password-stdin
+      - name: Push Docker image
+        if: ${{ github.ref == 'refs/heads/master' }}
+        run: |
+          docker push ghcr.io/grafana/gokrb5-test:${{ matrix.pair.tag }}
```
Comment (codebien) [inline .github/workflows/docker.yml:38]: Do we really need to push on each pull request? I guess we can do it only when we merge it.

Comment (joanlopez) [reply .github/workflows/docker.yml:38]: Sorry, the `if` I really wanted is:

```
if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

On pull requests I think the build (to make sure we detect any error on changes) is enough.

File: testenv/docker/krbhttp/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,6 @@ RUN cd /etc/yum.repos.d/ \
 
 RUN yum install -y \
   httpd \
-  mod_auth_kerb \
```
Comment (joanlopez) [inline testenv/docker/krbhttp/Dockerfile:16]: In theory with `mod_auth_gssapi` should be enough
 👉🏻 https://projects.theforeman.org/issues/32352

File: testenv/docker/krb5kdc-older/Dockerfile [hunk 3]
```diff
@@ -17,7 +17,7 @@ RUN yum install -y \
 
 ADD krb5-1.6.3.tar.gz /tmp
 RUN cd /tmp/krb5-1.6.3/src && \
-  ./configure && make && make install
+  ./configure && make CFLAGS="-fgnu89-inline" && make install
```
Comment (joanlopez) [inline testenv/docker/krb5kdc-older/Dockerfile:20]: Fixes the inlining/duplication issue with recent versions of `gcc` compiler:
👉🏻  https://bugzilla.redhat.com/show_bug.cgi?id=547153

**Review by codebien — APPROVED** [review]

**Pull request merged**